Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -889,15 +1012,42 @@ bool qSlicerSaveDataDialogPrivate::saveNodes()
                               QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
       if (answer == QMessageBox::No)
         {
+        if (snode)
+          {
+          snode->GetUserMessages()->AddMessage({vtkCommand::ErrorEvent,
+              qPrintable(tr("Cannot write data file: %1. Stopping save operation at user's request.").arg(file.absoluteFilePath()))});
```

Comment:
qPrintable is equivalent to str.toLocal8Bit().constData(), which is for writing to the console only. Application's code page is UTF-8, which may note be the same as the console. In all locations where you convert QString to std::string or char*, you need to https://doc.qt.io/qt-5/qstring.html#toStdString or https://doc.qt.io/qt-5/qstring.html#toUtf8, or use qUtf8Printable instead. Since we don't use qUtf8Printable anywhere in Slicer and there are alternatives, I would not start using it here.

Reply:
Very good.  I have switched to using `.toStdString()`.